Express/src/external/GoogleAPI.java

```diff
@@ -1,33 +1,184 @@
 package external;
 
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
 import db.DBConnection;
 import db.DBConnectionFactory;
 import entity.Location;
 import entity.Station;
+import rpc.HTTPHelper;
+import rpc.HTTPUtil;
 
 public class GoogleAPI {
-	// method 1: convert address string to latitude and longitude
-	public static double[] addr_to_lonlat(String address) {
-		double[] res = {37.7319638,-122.4508133};
-		return res;
+	public static String API_KEY = "AIzaSyARtu9LC3znDAtBj1dC4SOitTCxvTLfAFc";
+	public static String TEST_ADDRESS = "9500+Gilman+Dr,+La+Jolla,+CA+92093";
+	public static double carrier_speed = 12.0;     //Speed in meters
+	
+	//https://maps.googleapis.com/maps/api/geocode/json?address=9500+Gilman+Dr,+La+Jolla,+CA+92093&key=AIzaSyCMUv0b5DrWEeTUrHmO57WR-LpaDudaxwM
+	
+	//Method 1: Convert address string to latitude and longitude.
+	public static double[] addr_to_latlng(String address) {
+		try { 
+			String prefix = "https://maps.googleapis.com/maps/api/geocode/json?address=";
+			String keyHead = "&key=";
+			String request = prefix + address + keyHead + API_KEY;
+			//Concatenate URL
+			URL url = new URL(request);
+			HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+			
+			connection.setRequestMethod("GET");
+			connection.setRequestProperty("Content-Type", "application/json");
+			connection.setConnectTimeout(5000);
+			connection.setReadTimeout(5000);
+			
+			int status = connection.getResponseCode();
+			System.out.println("Status: "+status);
+			if (status != 200) {
+				return new double[2];
+			}
+			
+		
+			BufferedReader in = new BufferedReader(
+					  new InputStreamReader(connection.getInputStream()));
+			String inputLine;
+			StringBuilder content = new StringBuilder();
+			
+			
+			while ((inputLine = in.readLine()) != null) {
+			    content.append(inputLine);
+			}
+			in.close();				
+								
+			JSONObject obj = new JSONObject(content.toString());
+			if(!obj.isNull("results")) {
+				JSONArray results = obj.getJSONArray("results");
+				JSONObject item = results.getJSONObject(0);
+				JSONObject geometry = item.getJSONObject("geometry");
+				JSONObject location = geometry.getJSONObject("location");
+				double latitude = location.getDouble("lat");
+				double longitude = location.getDouble("lng");
+				double[] latlng = new double[2];
+				latlng[0] = latitude;
+				latlng[1] = longitude;
+				return latlng;
+			} else {
+				return new double[2];
+			}
+			
+		} catch(Exception e) {
+			e.printStackTrace();
+		}
+		return new double[2];
 	}
 	
-	// method 2: calculate time for spherical distance of two locations based on latitude and longitude
```

Comment:
general remove all the commented code before you push. the diff is supposed to have only comments and final code